### ISSUE TITLE
Rename-Many, paramter binding exception

### ISSUE THREAD

**Issue by plastikfan** [plastikfan/Loopz#73]
Whilst building the following test, a parameter binding exception occured. There should be no circumstance in which this can happen. Invalid user invocation should be accounted for and handled gracefully.

```powershell
  Context 'given: Diagnose enabled' {
    Context 'MoveToAnchor' {
      Context 'and: Source matches with Named Captures' {
        Context 'and: Copy matches' {
          Context 'and: Anchor matches' {
            It 'should: do rename; move Pattern match with Copy capture' -Tag 'Current' {
              $script:expected = @{
                'loopz.application.t1.log' = 'application.BEGIN-.t1-application.-loopz.-END.log';
                'loopz.application.t2.log' = 'application.BEGIN-.t2-application.-loopz.-END.log';
                'loopz.data.t1.txt'        = 'data.BEGIN-.t1-data.-loopz.-END.txt';
                'loopz.data.t2.txt'        = 'data.BEGIN-.t2-data.-loopz.-END.txt';
                'loopz.data.t3.txt'        = 'data.BEGIN-.t3-data.-loopz.-END.txt';
              }

              [string]$pattern = '^(?<header>[\w]+)\.';
              [string]$anchor = '\.(?<tail>t\d)';
              [string]$copy = '(?<body>[\w]+)\.';
              [string]$paste = '.BEGIN-${_a}-${_c}-${header}-END';

              Get-ChildItem -File -Path $directoryPath | Rename-Many -File `
                -Pattern $pattern -Copy $copy -Anchor $anchor -Relation 'after' -Paste $paste -WhatIf -Diagnose;
            }
          }
        }
      }
    } # MoveToAnchor
```

generates the following output:

```
[🎯] ----------------------------------------------------------------------------------------------------- [ Rename ] ---
[-] Rename-Many.given: Diagnose enabled.MoveToAnchor.and: Source matches with Named Captures.and: Copy matches.and: Anchor matches.should: do rename; move Pattern match with Copy capture 28ms (27ms|1ms)
 ParameterBindingValidationException: Cannot bind argument to parameter 'MatchObject' because it is null.
 CmdletInvocationException: Cannot bind argument to parameter 'MatchObject' because it is null.
 MethodInvocationException: Exception calling "Invoke" with "4" argument(s): "Cannot bind argument to parameter 'MatchObject' because it is null."
 at <ScriptBlock>, C:\Users\Plastikfan\dev\github\PoSh\Loopz\Elizium.Loopz\Output\Elizium.Loopz\Elizium.Loopz.psm1:3377
 at Write-HostFeItemDecorator, C:\Users\Plastikfan\dev\github\PoSh\Loopz\Elizium.Loopz\Output\Elizium.Loopz\Elizium.Loopz.psm1:3381
 at <ScriptBlock>, C:\Users\Plastikfan\dev\github\PoSh\Loopz\Elizium.Loopz\Output\Elizium.Loopz\Elizium.Loopz.psm1:255
 at Invoke-ForeachFsItem<Process>, C:\Users\Plastikfan\dev\github\PoSh\Loopz\Elizium.Loopz\Output\Elizium.Loopz\Elizium.Loopz.psm1:829
 at Rename-Many<End>, C:\Users\Plastikfan\dev\github\PoSh\Loopz\Elizium.Loopz\Output\Elizium.Loopz\Elizium.Loopz.psm1:2843
 at <ScriptBlock>, C:\Users\Plastikfan\dev\github\PoSh\Loopz\Elizium.Loopz\Tests\Public\Rename-Many.tests.ps1:515
```

**Comment by plastikfan** [comment]
This is caused by this line inside Move-Match:

```powershell
          [string]$replaceWith, $null, $copyMatch = Split-Match `
            -Source $patternRemoved -PatternRegEx $Copy `
            -Occurrence ($PSBoundParameters.ContainsKey('CopyOccurrence') ? $CopyOccurrence : 'f') `
            -CapturedOnly;
```

In particular, if you specifiy **-CapturedOnly**, then the match object is not returned, so in this case, since we need the match object, CapturedOnly should not be specified.

**Comment by plastikfan** [comment]
This is such a small fix, that it should just be fixed by issue #71.
